pkg/handler/handler_test.go

```diff
@@ -121,6 +121,48 @@ var rawPodWithVolume = []byte(`
 }
 `)
 
+var rawPodWithIAMTokenVolume = []byte(`
```

Comment:
Could you make this test more representative of what this patch is doing?
The idea is that  `rawPodWithIAMTokenVolume` should have a `volume` declared but `env` and `volumeMounts` should not be present. 

`validPatchIfIAMTokenVolumePresent` would then prove that with this patch we are adding `env` and `volumeMounts` correctly now.

Reply:
Thanks for the review! I think the main behavior we want to test is that`volumeMounts` and `env` are added once, _and only once_, regardless of if the `volume` is already present.

`rawPodWithIAMTokenVolume` (as it stands) tests the case where 1) `volume` present, 2) `volumeMount` present, no duplicate `volumeMount` added.

I can see how this can be mis-leading, so I've separated the cases for whether a `volumeMount` is present or not.